### PR TITLE
Prevent security issues from preload media option

### PR DESCRIPTION
**Pull request by janicklas-ralph**
Handles the security review feedback. Validates the media option to ensure there is no unsafe HTML being injected

### CODE REVIEW

File: packages/critters/src/css.js [hunk 1]
```diff
@@ -191,3 +192,64 @@ function filterSelectors(predicate) {
...
+ * @returns {boolean}
+ *
+ * This function performs a basic media query validation
+ * to ensure the values passed as part of the 'media' config
+ * is HTML safe and does not cause any injection issue
+ */
+export function validateMediaQuery(query) {
+  console.log(query);
+  const mediaTree = mediaParser(query);
+  const nodeTypes = ['media-type', 'keyword', 'media-feature'];
```
Comment (alan-agius4) [inline packages/critters/src/css.js:240]: NIT: consider using Set where possible for faster lookups.

File: packages/critters/src/css.js [hunk 2]
```diff
@@ -191,3 +192,64 @@ function filterSelectors(predicate) {
...
+  const stack = [mediaTree];
+
+  while (stack.length > 0) {
+    const node = stack.pop();
+
+    if (nodeTypes.includes(node.type) && !validateMediaType(node)) {
+      return false;
+    }
+
+    stack.push(...(node.nodes || []));
```
Comment (alan-agius4) [inline packages/critters/src/css.js:251]:
```suggestion
    if (node.nodes?.length) {
      stack.push(...node.nodes);
    }
```

File: packages/critters/package.json [hunk 3]
```diff
@@ -52,6 +52,7 @@
     "domhandler": "^5.0.2",
     "htmlparser2": "^8.0.2",
     "postcss": "^8.4.23",
+    "postcss-media-query-parser": "^0.2.3",
```
Comment (alan-agius4) [inline packages/critters/package.json:55]: In general I would be hesitant in using a package that was last released 7 years ago.

**Review by alan-agius4 — APPROVED** [review]
Couple of NITs

File: packages/critters/package.json (hunk 3 above)
Comment (janicklas-ralph) [reply packages/critters/package.json:55]: Makes sense. I'll try to find a different library/approach later. For now I think this covers our basic usecase.

**Review by alan-agius4 — APPROVED** [review]
LGTM

**Pull request merged**